### ISSUE TITLE
Source Xero: includes Journals, Quotes, and Reports data streams

### ISSUE THREAD

**Issue by 0zero** [airbytehq/airbyte#31395]
### What area the feature impact?

Connectors

### Relevant Information

I would like to add the [Journals](https://developer.xero.com/documentation/api/accounting/journals), [Quotes](https://developer.xero.com/documentation/api/accounting/quotes) and [Reports](https://developer.xero.com/documentation/api/accounting/reports#profit-and-loss) data streams to the `source-xero` connector. 

We'd would like to use the additional data streams to do some reporting in our BI tools and need the data in our data warehouse but with the available connector this isn't possible which is frustrating to have to go back and forth exporting data to csv and uploading them in order to do things with them.

**Comment by aidanlister** [comment]
I've written some python to fetch Profit and Loss data, would that be helpful?

```
def get_contact_account_numbers(accounting_api: Any, xero_tenant_id: str) -> dict[str, str]:
    # Grab all the contacts
    account_codes = {}
    contacts = accounting_api.get_contacts(xero_tenant_id, where="IsCustomer==True")
    for contact in contacts.contacts:
        account_codes[contact.contact_id] = contact.account_number
    return account_codes


def get_chart_of_accounts(accounting_api: Any, xero_tenant_id: str) -> list[dict]:
    accounts = accounting_api.get_accounts(
        xero_tenant_id,
    )
    rows = []
    for account in accounts.accounts:
        rows.append(
            {
                "last_updated": arrow.now(UPTICK_HQ_TIMEZONE).isoformat(),
                "name": account.name,
                "code": account.code,
            }
        )
    return rows


def get_profit_and_loss(
    accounting_api: Any,
    xero_tenant_id: str,
    chart_of_accounts: list[dict[Any, Any]],
    months: int = 12,
    tracking_category: str | None = None,
    header_labels: list[str] | None = None,
) -> list[dict]:
    START_SPAN = arrow.now(UPTICK_HQ_TIMEZONE).floor("month").shift(months=-months)
    END_SPAN = arrow.now(UPTICK_HQ_TIMEZONE).ceil("month")

    tracking_category_params = {"tracking_category_id": tracking_category} if tracking_category else {}

    if header_labels is None:
        header_labels = []

    account_to_account_and_code_map = {}
    for account in chart_of_accounts:
        account_to_account_and_code_map[account["name"]] = f"{account['code']} - {account['name']}"

    data = []

    """ Because Xero's API is a giant piece of shit, to grab an accurate P&L you need to run this API call 11 times.
    https://community.xero.com/developer/discussion/100915956
    """
    for from_date, to_date in arrow.Arrow.span_range("month", START_SPAN.datetime, END_SPAN.datetime):
        timeframe = "MONTH"
        report = accounting_api.get_report_profit_and_loss(
            xero_tenant_id,
            from_date.date(),
            to_date.date(),
            timeframe,
            **tracking_category_params,
        )

        headers = []
        for xero_row in report.reports[0].rows:
            if xero_row.row_type == RowType.HEADER:
                headers = [cell.value for cell in xero_row.cells]

            if xero_row.row_type == RowType.SECTION:
                for row in xero_row.rows:
                    section_values = {headers[i]: cell.value for i, cell in enumerate(row.cells)}
                    data.append(
                        {
                            "account_code": account_to_account_and_code_map.get(section_values[""], section_values[""]),
                            **{k: section_values.get(k, "") for k in header_labels},
                            **{
                                "total": section_values.get(to_date.strftime("%d %b %y"), "")
                                if not tracking_category
                                else section_values.get("Total", "")
                            },
                            **{"date": to_date.date().isoformat()},
                            **{"updated": arrow.now(UPTICK_HQ_TIMEZONE).isoformat()},
                        }
                    )

    return data
```

We call it like so:

```
    # ... get the account codes
    chart_of_accounts = get_chart_of_accounts(
        accounting_api,
        xero_tenant_id,
    )
    write_to_sheet(sheet, "Account Codes", chart_of_accounts, clear_sheet_first=True)
    logger.info(" ... done with chart of accounts")

    # ... get the P&L
    logger.info(f"Exporting profit and loss ({MONTHS} months) ...")
    profit_and_loss = get_profit_and_loss(
        accounting_api,
        xero_tenant_id,
        chart_of_accounts,
        months=MONTHS,
        tracking_category=GEOGRAPHY_TRACKING_CATEGORY_GUIDS[ORG_LOOKUP_KEY],
        header_labels=PROFIT_AND_LOSS_HEADERS[ORG_LOOKUP_KEY],
    )
    write_to_sheet(sheet, "Live Profit and Loss", profit_and_loss, clear_sheet_first=True)
    logger.info(" ... done with P&L")
```

Anyone keen to collaborate on getting this into AirByte?

**Comment by blox-si-ritchie** [comment]
We're very excited to use the Xero connector. I believe the main endpoint we need is the Journals endpoint.
Can we collaborate to get this in. Happy to help with getting the code written if someone is lined up to review it.

**Comment by octavia-squidington-iii** [comment]
At Airbyte, we seek to be clear about the project priorities and roadmap. This issue has not had any activity for 180 days, suggesting that it's not as critical as others. It's possible it has already been fixed. It is being marked as stale and will be closed in 20 days if there is no activity. To keep it open, please comment to let us know why it is important to you and if it is still reproducible on recent versions of Airbyte.
